src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php

```diff
@@ -377,7 +379,35 @@ public function forget($key)
      */
     public function flush()
     {
-        throw new RuntimeException('DynamoDb does not support flushing an entire table. Please create a new table.');
+        try {
+            // Delete the table
+            $this->dynamoDb->deleteTable(['TableName' => $this->table]);
+        } catch (HttpException $e) {
+            if ($e->getCode() !== 404) {
+                // Any error but "table not found"
+                throw new \RuntimeException('Could not flush DynamoDb cache. Table could not be deleted.', $e->getCode(), $e);
+            }
+        }
+
+        // Wait until table is removed
+        if (!$this->dynamoDb->tableNotExists(['TableName'=>$this->table])->isSuccess()) {
+            throw new \RuntimeException('Could not flush DynamoDb cache. Table could not be deleted.');
+        }
+
+        // Create a new table
+        $this->dynamoDb->createTable([
```

Comment:
what 's about crossed concurrency. should we use a kind of lock somewhere?

Reply:
Hm.. But how?

The only shared resource we know about is `DynamoDb`, and Im removing it. =)

FYI: https://github.com/async-aws/aws/pull/588

Reply:
I'm not sure this worse a so complex thing.. 
Let keep it simple. Maybe we should reduce the time between delete and create table (ie. by using `$response->wait(500, 5)` by default, the TableDoesNotExists waiter poll every 20 seconds which seems huge

Reply:
Thank you. I've updated the PR